packages/jest-snapshot/src/index.js

```diff
@@ -92,7 +92,7 @@ const toMatchSnapshot = function(received: any, testName?: string) {
 
     report = () =>
       `${RECEIVED_COLOR('Received value')} does not match ` +
-      `${EXPECTED_COLOR('stored snapshot ' + count)}.\n\n` +
+      `${EXPECTED_COLOR(`stored snapshot "${result.key}"`)}.\n\n` +
```

Comment:
destructure `key`?

Reply:
I guess my above comment got swallowed by subsequent commits.
I agree with wrapping the output in double braces, but the toggling behavior (quote or not to quote) seems a bit awkward to me. I would rather favor to have the complete snapshot key on the output:

```
Received value does not match stored snapshot "failing named snapshot: snapname 1".
```

This would speed up finding the right one in the snap file, as well. Thus I updated  would require changes to State.js as well.